include/nebula/sclient/StorageClient.h

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2018 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <thread>
+#include <unordered_map>
+#include <vector>
+
+#include "ScanEdgeIter.h"
+#include "common/datatypes/HostAddr.h"
+#include "common/thrift/ThriftTypes.h"
+#include "nebula/mclient/MetaClient.h"
+#include "nebula/sclient/ScanEdgeIter.h"
+
+namespace folly {
+class IOThreadPoolExecutor;
+template <class T>
+class Promise;
+}  // namespace folly
+
+namespace nebula {
+
+namespace thrift {
+
+template <class ClientType>
+class ThriftClientManager;
+
+}  // namespace thrift
+
+namespace meta {
+namespace cpp2 {
+
+class MetaServiceAsyncClient;
+
+}  // namespace cpp2
+}  // namespace meta
+
+namespace storage {
+namespace cpp2 {
+
+class GraphStorageServiceAsyncClient;
+class ScanCursor;
+class ScanEdgeRequest;
+class ScanEdgeResponse;
+
+}  // namespace cpp2
+}  // namespace storage
+
+#define DEFAULT_LIMIT 1000
+#define DEFAULT_START_TIME 0
+#define DEFAULT_END_TIME std::numeric_limits<int64_t>::max()
+
+class StorageClient {
+  friend struct ScanEdgeIter;
+
+ public:
+  explicit StorageClient(const std::vector<std::string>& metaAddrs);
+
+  ~StorageClient();
+
+  std::vector<PartitionID> getParts(const std::string& spaceName);  // plato needed
+
+  ScanEdgeIter scanEdgeWithPart(std::string spaceName,
+                                int32_t partID,
+                                std::string edgeName,
+                                std::vector<std::string> propNames,
+                                int64_t limit = DEFAULT_LIMIT,
```

Comment:
Don't limit by default, set it to max number.

Reply:
Why? If the limit is too big, it may scan out all data of a huge partition and thus cause the storaged OOM.

Reply:
It's default.

Reply:
OK